lib/lettering/font.py

```diff
@@ -206,13 +214,20 @@ def _render_glyph(self, glyph, position, character, last_character):
         """
 
         node = deepcopy(glyph.node)
-
         if last_character is not None:
-            position.x += self.letter_spacing + self.kerning_pairs.get(last_character + character, 0) * PIXELS_PER_MM
-
+            if self.version == "2":
+                position.x += glyph._min_x - self.kerning_pairs_px.get(last_character + character, 0)  
```

Comment:
The prefix `_` in `_min_x` is a way of saying "this is designed only to be used by methods in the Glyph class", a.k.a. a private member variable.  I didn't think we'd need `min_x` outside the Glyph class when I wrote it.  Now it's clear that we need it, so I think we should change the name of the member variable to `min_x`, without the underscore prefix.  Same for `max_x` and the others.

Reply:
Ok. I have to modify or is it you?

Reply:
Go right ahead.  No need to ask me first before changing things.  :)

Reply:
I think it would make sense for it to be part of this pull request.  I might have a chance to do it this evening, or you can -- whoever gets to it first :)

Reply:
Done!

Reply:
Here is something I don't understand: why has this changed to subtracting the `self.kerning_pairs_px` value?  Is this another thing I don't understand about fonts?

Reply:
@lexelby 
The values ​​entered in <hkern g1 = "T" g2 = "s" k = "7" /> are values ​​to be subtracted. I didn't invent the rule: https://www.w3.org/TR/SVG11/fonts.html#HKernElement
If the spacing between letters needs to be made bigger there will be I imagine <hkern g1 = "T" g2 = "s" k = "-7" />. But I think it doesn't happen often.
Did I understand your question correctly?

Reply:
And concerning min_x: I add it before moving the letter because it is to take into account the margin in the drawing of the letter. With respect to point 0 the letter can start at 5 or -5. The letters have a defined place in the drawing that's important.
Then to calculate the position of x for the next letter I have to remove the min_x margin because the horizontal adv is calculated from point 0 of the drawing. You have to look at the fontforge software to understand what is going. 
You can find some frontforge print screens here which explains a bit : =on.https://github.com/inkstitch/inkstitch/issues/809

Reply:
Okay!  Then let's have our kerning work this way too, both for `kerning_pairs` and `kerning_pairs_px`.

I wonder if we should just make `kerning_pairs` be in pixels.....